plugins/org.komodo.spi/src/org/komodo/spi/repository/KomodoObject.java

```diff
@@ -142,6 +142,11 @@ Property getProperty( final UnitOfWork transaction,
     String[] getPropertyNames( final UnitOfWork transaction ) throws KException;
 
     /**
+     * @return a unique identifier for the object class
+     */
+    int getTypeId();
```

Comment:
Don't see a problem with this method of id. However, this commit only implements for AccessPattern[Impl]. Should it not implement for all relational types?